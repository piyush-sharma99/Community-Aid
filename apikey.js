/*
 *  ClassName: apikey.js
 *
 *  Date: 28/03/2021
 *
 * @author Piyush Sharma, X17342356
 *
 * @reference https://reactnative.dev/docs/activityindicator
 * @reference https://www.udemy.com/course/react-native-the-practical-guide/
 * @reference https://www.youtube.com/watch?v=8zT6CYu0iYQ
 *
 */

export const firebaseConfig = {
  apiKey: "AIzaSyDGliQfMfKpXUllG5ZPyClJL6C8VgFKXo8",
  authDomain: "community-aid.firebaseapp.com",
  databaseURL:
    "https://community-aid-default-rtdb.europe-west1.firebasedatabase.app",
  projectId: "community-aid",
  storageBucket: "community-aid.appspot.com",
  messagingSenderId: "529896701339",
  appId: "1:529896701339:web:1c30369d39f8654b01353e",
  measurementId: "G-PPD28P0PC6",
};
