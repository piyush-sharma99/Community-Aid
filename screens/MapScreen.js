import React, { useState, useEffect } from 'react';
import { StyleSheet, Text, View, TextInput, Button, TouchableOpacity } from 'react-native';
import MapView, {Marker, Circle} from 'react-native-maps';
import {Card} from 'react-native-paper';
import * as Location from 'expo-location';



const MapScreen = props => {
  const [location, setLocation] = useState(null);
  const [errorMsg, setErrorMsg] = useState(null);
  const [radius, setRadius] = useState(1000);
  const [updateRadius, setUpdateRadius] = useState(1000);

  


    useEffect(() => {
      (async () => {
        let { status } = await Location.requestPermissionsAsync();
        if (status !== 'granted') {
          setErrorMsg('Permission to access location was denied');
          return;
        }
  
        let location = await Location.getCurrentPositionAsync({});
        setLocation(location);
      })();
    }, []);
  
    let text = 'Waiting..';
    let latitude = -9.062691;
    let longitude = 53.270962;
    if (errorMsg) {
      text = errorMsg;
    } else if (location) {
      text = JSON.stringify(location);
      longitude = location.coords.longitude;
      latitude = location.coords.latitude;
      
    }
    console.log(longitude);
    console.log(latitude);

    return (
    <View style={styles.screen} >
        <View style={styles.mapView}>
        <MapView style={styles.map} 
        region={{
          latitude: latitude,
          longitude: longitude,
          latitudeDelta: 0.02,
          longitudeDelta: 0.30,
        }}> 
        <Marker coordinate = {{latitude: latitude, longitude:longitude}}/>
        <Circle
        center={{latitude: latitude, longitude:longitude}}
        radius={updateRadius}
        fillColor={'rgba(43, 98, 227, 0.2)'}
         />
        </MapView>
        </View>

        <Card style={styles.cardView}>
        <View style={styles.inputView} >
          <TextInput  
            style={styles.inputText}
            placeholder="Radius in kilometers eg (3)..." 
            placeholderTextColor="#003f5c"
            onChangeText={(radius) => setRadius(parseInt(radius))}
            />
            </View>
            <TouchableOpacity style={styles.Btn} onPress = {()=>setUpdateRadius(radius * 1000)}>
             <Text style={styles.subText} >Change radius</Text>
        </TouchableOpacity>
            </Card>
    </View>
    );
};

MapScreen.navigationOptions = {
    headerTitle: 'MAP ',
    headerStyle: {
    backgroundColor: '#2E86C1'
    },
    headerTintColor:"white"
    };
    

const styles = StyleSheet.create({
    screen: {
        width:'100%',
        height:'100%',
        flex: 1,
        backgroundColor: '#2E86C1',
        alignItems: 'center',
        justifyContent: 'center'
    },
    mapView: {
        width:'95%',
        height:'65%',
        alignItems: 'center',
        justifyContent: 'center',
        borderWidth: 3,
        borderColor: "#fb5b5a",
        backgroundColor: "#61dafb",
        marginBottom: 20,
        marginTop:-30,
        shadowColor: 'black',
        shadowOpacity: 2,
        elevation: 10
        
    },
    map: {
        width:'100%',
        height:'100%',
        backgroundColor: "#61dafb",

    },
    cardView: {
        width:'95%',
        height:'25%',
        backgroundColor:"#ffffff",
        shadowColor: 'black',
        alignItems: 'center',
        borderRadius:25,
        shadowOpacity: 2,
        elevation: 10,
    },
    inputView:{
        marginTop:30,
        width:250,
        backgroundColor:"#FDFEFE",
        borderRadius:25,
        height:50,
        justifyContent:"center",
        padding:20,
        borderWidth: 4,
        borderColor: "#fb5b5a",
        shadowColor: 'black',
        borderRadius:25,
      },

      inputText:{
        height:50,
        color:"black"
      },

      Btn:{
        width:250,
        backgroundColor:"#fb5b5a",
        borderRadius:25,
        height:50,
        alignItems:"center",
        justifyContent:"center",
        marginTop:20,
        marginBottom:5
      },

      subText:{
        color:"white",
        fontSize:20
      },
    
});

export default MapScreen;