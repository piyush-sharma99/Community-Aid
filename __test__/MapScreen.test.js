/*
 *  ClassName: MapScreen.test.js
 *
 *  Date: 28/03/2021
 *
 * @author Piyush Sharma, X17342356
 *
 * @reference https://docs.expo.io/
 */
import React from "react";
import MapScreen from "../screens/MapScreen";
import { firebaseConfig } from "../apikey";
import { cleanup } from "@testing-library/react-native";
import renderer from "react-test-renderer";
import * as firebase from "firebase";

firebase.initializeApp(firebaseConfig);

afterEach(cleanup);

test("Map renders correctly", () => {
  const tree = renderer.create(<MapScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
