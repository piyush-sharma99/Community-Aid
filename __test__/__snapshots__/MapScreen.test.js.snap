// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map renders correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "height": "100%",
      "width": "100%",
    }
  }
>
  <Image
    source={1}
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": "100%",
          "width": "100%",
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "height": "100%",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fb5b5a",
          "borderColor": "white",
          "borderRadius": 25,
          "borderWidth": 3,
          "elevation": 5,
          "marginBottom": 10,
          "marginTop": 20,
          "padding": 5,
          "shadowColor": "black",
          "shadowOffset": Object {
            "height": 3,
            "width": 3,
          },
          "shadowOpacity": 2,
          "shadowRadius": 5,
          "width": "95%",
        }
      }
    >
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flexGrow": 1,
            "flexShrink": 1,
          }
        }
      >
        <View
          index={0}
          siblings={
            Array [
              "View",
            ]
          }
          style={
            Object {
              "backgroundColor": "#FDFEFE",
              "borderColor": "#2E86C1",
              "borderRadius": 25,
              "borderWidth": 4,
              "height": 50,
              "justifyContent": "center",
              "marginTop": 10,
              "padding": 20,
              "shadowColor": "black",
              "width": 250,
            }
          }
          total={2}
        >
          <TextInput
            allowFontScaling={true}
            onChangeText={[Function]}
            placeholder="Radius in kilometers eg (3)..."
            placeholderTextColor="#003f5c"
            rejectResponderTermination={true}
            style={
              Object {
                "color": "black",
                "height": 50,
              }
            }
            underlineColorAndroid="transparent"
          />
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#2E86C1",
              "borderColor": "white",
              "borderRadius": 10,
              "borderWidth": 2,
              "height": 50,
              "justifyContent": "center",
              "marginBottom": 10,
              "marginTop": 20,
              "opacity": 1,
              "width": 250,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 20,
              }
            }
          >
            Change radius
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#61dafb",
          "elevation": 10,
          "height": "75%",
          "justifyContent": "center",
          "marginTop": 10,
          "width": "100%",
        }
      }
    >
      <AIRMap
        handlePanDrag={false}
        initialRegion={null}
        onChange={[Function]}
        onLayout={[Function]}
        onMapReady={[Function]}
        onMarkerPress={[Function]}
        region={
          Object {
            "latitude": -9.062691,
            "latitudeDelta": 0.02,
            "longitude": 53.270962,
            "longitudeDelta": 0.3,
          }
        }
        style={
          Object {
            "backgroundColor": "#61dafb",
            "height": "100%",
            "width": "100%",
          }
        }
      >
        <AIRMapMarker
          coordinate={
            Object {
              "latitude": -9.062691,
              "longitude": 53.270962,
            }
          }
          onPress={[Function]}
          stopPropagation={false}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "position": "absolute",
              },
              undefined,
            ]
          }
          title="Your location"
        >
          <Text />
        </AIRMapMarker>
        <AIRMapCircle
          center={
            Object {
              "latitude": -9.062691,
              "longitude": 53.270962,
            }
          }
          fillColor="rgba(43, 98, 227, 0.2)"
          radius={1000}
          strokeColor="#000"
          strokeWidth={1}
        />
      </AIRMap>
    </View>
  </View>
</View>
`;
